src/pkg/services/m365/api/lists.go

```diff
@@ -658,6 +662,44 @@ func hasHyperLinkFields(additionalData map[string]any) (map[string]any, string,
 	return nil, "", false
 }
 
+func hasMetadataFields(additionalData map[string]any) ([]map[string]any, string, bool) {
+	for fieldName, value := range additionalData {
+		switch valType := reflect.TypeOf(value).Kind(); valType {
```

Comment:
This var assignment feels redundant.  Is this the same result?
```suggestion
		switch reflect.TypeOf(value).Kind() {
```